app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsViewModel.kt

```diff
@@ -0,0 +1,505 @@
+                    Log.d(TAG, "Saving visible ID: ${action.visibleId}")
+                    accountManager.activeAccount?.let { account ->
+                        account.lastNotificationId = action.visibleId
+                        accountManager.saveAccount(account)
+                    }
+                }
+        }
+
+        // Set initial status display options from the user's preferences.
+        //
+        // Then collect future preference changes and emit new values in to
+        // statusDisplayOptions if necessary.
+        statusDisplayOptions = MutableStateFlow(
+            StatusDisplayOptions.from(
+                preferences,
+                accountManager.activeAccount!!
+            )
+        )
+
+        viewModelScope.launch {
+            eventHub.events.asFlow()
+                .filterIsInstance<PreferenceChangedEvent>()
+                .filter { StatusDisplayOptions.prefKeys.contains(it.preferenceKey) }
+                .map {
+                    statusDisplayOptions.value.make(
+                        preferences,
+                        it.preferenceKey,
+                        accountManager.activeAccount!!
+                    )
+                }
+                .collect {
+                    statusDisplayOptions.emit(it)
+                }
+        }
+
+        // Handle UiAction.ClearNotifications
+        viewModelScope.launch {
+            uiAction.filterIsInstance<FallibleUiAction.ClearNotifications>()
+                .collectLatest {
+                    try {
+                        repository.clearNotifications().apply {
+                            if (this.isSuccessful) {
+                                repository.invalidate()
+                            } else {
+                                uiError.emit(UiError.make(HttpException(this), it))
+                            }
+                        }
+                    } catch (e: Exception) {
+                        ifExpected(e) { uiError.emit(UiError.make(e, it)) }
+                    }
+                }
+        }
+
+        // Handle NotificationAction.*
+        viewModelScope.launch {
+            uiAction.filterIsInstance<NotificationAction>()
+                .debounce(DEBOUNCE_TIMEOUT_MS)
+                .collect { action ->
+                    try {
+                        when (action) {
+                            is NotificationAction.AcceptFollowRequest ->
+                                timelineCases.acceptFollowRequest(action.accountId)
+                            is NotificationAction.RejectFollowRequest ->
+                                timelineCases.rejectFollowRequest(action.accountId)
+                        }
+                        uiSuccess.emit(NotificationActionSuccess.from(action))
+                    } catch (e: Exception) {
+                        ifExpected(e) { uiError.emit(UiError.make(e, action)) }
+                    }
+                }
```

Comment:
I think accepting or rejecting a follow request should make the notification go away, currently I can accept multiple times.

Reply:
Fixed.